src/main/java/io/github/nscuro/versatile/VersUtils.java

```diff
@@ -84,7 +84,8 @@ public static Vers versFromGhsaRange(final String ecosystem, final String rangeE
      * @throws VersException            When the produced {@link Vers} is invalid
      * @throws InvalidVersionException  When any version in the range is invalid according to the inferred {@link VersioningScheme}
      */
-    public static Vers versFromOsvRange(final String type, final String ecosystem, final List<Map.Entry<String, String>> events) {
+    public static Vers versFromOsvRange(final String type, final String ecosystem,
+                                        final List<Map.Entry<String, String>> events, final String databaseSpecificLastKnownAffected) {
```

Comment:
`last_known_affected` is a GitHub-specific thing IIRC. I think from an API perspective for versatile, it would make more sense to accept a `Map<String, Object> databaseSpecific` parameter here.

As per [OSV spec](https://ossf.github.io/osv-schema/#affecteddatabase_specific-field):

> The `affected` object’s `database_specific` field is a JSON object holding additional information about the vulnerability as defined by the database from which the record was obtained. 

Given the `Map<String, Object>`, we can then check for an entry with key `last_known_affected`, and apply the respective logic. This way, the caller of `versFromOsvRange` doesn't have to care whether `last_known_affected` is set or not. And we avoid having to add new method parameter for every new database-specific field that pops up.

Reply:
Shouldn't it be `Map.Entry<String, String>` instead then? To parse string values against keys in the jsonObject? 
```
"database_specific": {
  "last_known_affected_version_range": "<= 29.0"
}
```

Reply:
Generally agree, however the OSV spec does not guarantee that all fields within the `database_specific` object are strings. It might as well be:

```json
{
  "database_specific": {
    "foo": {
      "bar": "baz"
    }
}
```

For us, it just means that we need to check the type of the value before using it, e.g.

```java
if (databaseSpecific.get("last_known_affected_version_range") instanceof final String lastKnownAffected) {
    // ...
}
```